app/src/main/java/org/oppia/app/home/continueplaying/ContinuePlayingFragmentPresenter.kt

```diff
@@ -0,0 +1,94 @@
+package org.oppia.app.home.continueplaying
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.Transformations
+import org.oppia.app.R
+import org.oppia.app.databinding.ContinuePlayingFragmentBinding
+import org.oppia.app.fragment.FragmentScope
+import org.oppia.app.model.OngoingStoryList
+import org.oppia.app.model.PromotedStory
+import org.oppia.app.topic.RouteToStoryListener
+import org.oppia.domain.topic.TopicListController
+import org.oppia.util.data.AsyncResult
+import javax.inject.Inject
+
+/** The presenter for [ContinuePlayingFragment]. */
+@FragmentScope
+class ContinuePlayingFragmentPresenter @Inject constructor(
+  private val activity: AppCompatActivity,
+  private val fragment: Fragment,
+  private val topicListController: TopicListController
+) {
+
+  private val routeToStoryListener = activity as RouteToStoryListener
+
+  private lateinit var binding: ContinuePlayingFragmentBinding
+
+  private lateinit var ongoingListAdapter: OngoingListAdapter
+
+  private val itemList: MutableList<ContinuePlayingViewModel> = ArrayList()
+
+  fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View? {
+    binding = ContinuePlayingFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false)
+
+    ongoingListAdapter = OngoingListAdapter(itemList)
+
+    binding.ongoingStoryRecyclerView.apply {
+      adapter = ongoingListAdapter
+    }
+    binding.let {
+      it.lifecycleOwner = fragment
+    }
+
+    subscribeToOngoingStoryList()
+
+    return binding.root
+  }
+
+  private val ongoingStoryListSummaryResultLiveData: LiveData<AsyncResult<OngoingStoryList>> by lazy {
+    topicListController.getOngoingStoryList()
+  }
+
+  private fun subscribeToOngoingStoryList() {
+    getAssumedSuccessfulOngoingStoryList().observe(fragment, Observer<OngoingStoryList> { it ->
+      if (it.recentStoryCount > 0) {
+        val recentSectionTitleViewModel = SectionTitleViewModel()
+        recentSectionTitleViewModel.sectionTitleText = activity.getString(R.string.ongoing_story_last_week)
+        itemList.add(recentSectionTitleViewModel)
+        for (promotedStory in it.recentStoryList) {
+          val ongoingStoryViewModel = OngoingStoryViewModel(promotedStory, fragment as OngoingStoryClickListener)
+          itemList.add(ongoingStoryViewModel)
+        }
+      }
+
+      if (it.olderStoryCount > 0) {
+        val olderSectionTitleViewModel = SectionTitleViewModel()
+        olderSectionTitleViewModel.sectionTitleText = activity.getString(R.string.ongoing_story_last_month)
+        if(itemList.size>0){
```

Comment:
Nit: need to have spacing around ``>``, before ``(``, and after ``)``. However we can simplify this to: ``if (itemList.isNotEmpty()) {``

Reply:
Resolved.